src/Diagnostics/dump_tendencies.jl

```diff
@@ -0,0 +1,251 @@
+"""
+    setup_dump_tendencies_diagnostics(
+        ::ClimateMachineConfigType,
+        interval::String,
+        out_prefix::String;
+        writer = NetCDFWriter(),
+        interpol = nothing,
+    )
+
+Create the "DumpTendencies" `DiagnosticsGroup` which contains all the
+tendencies for the simulation's `BalanceLaw`. These are collected on
+an interpolated grid (`interpol` _must_ be specified).
```

Comment:
Should we include a warning that this is extremely expensive?

Reply:
We should also note that this is more of a physics-debugging tool rather than numerics debugging, since these tendency evaluations will not match up exactly with what the RHS evaluation that the solver sees.

Reply:
Please post a suggestion for the text and I'll add it as is?